modeling/model.go

```diff
@@ -18,98 +18,73 @@ package modeling
 	}
 	model := &Model{
-		Region:        region,
 		SliceCapacity: 100,
 		alg:           alg,
 		simulator:     sim,
 	}
-	slices, err := model.alg.CreateSliceGroups(model.Region)
-	if err != nil {
-		return nil, err
-	}
-	model.slices = slices
 	return model, nil
 }
 
-// StartSimulation based on the zones(Region) and EndpointSliceGroups
-func (m *Model) StartSimulation() error {
-	simResult, err := m.simulator.Simulate(m.Region, m.slices)
+// UpdateRegion updates the region of the model, this is used to run the
+// algorithm on different zone inputs
+func (m *Model) UpdateRegion(zones []Zone) error {
+	region, err := createRegionInfo(zones)
+	if err != nil {
+		return err
+	}
+	slices, err := m.alg.CreateSliceGroups(region)
 	if err != nil {
 		return err
 	}
-	m.Results = append(m.Results, simResult)
+	m.region = region
+	m.slices = slices
 	return nil
 }
 
-// PrintLastResult prints the summary of the last simulation result
-func (m *Model) PrintLastResult() error {
-	numberOfSlices := m.GetNumberOfEndpointSlices()
-	lastResult := m.Results[len(m.Results)-1]
-	fmt.Printf("%% in-zone traffic \t %.2f%%\n", lastResult.InZoneTraffic*100)
-	fmt.Printf("# of endpoint slices\t %v\n", numberOfSlices)
-	fmt.Printf("# of endpoints\t %d\n", m.Region.totalEndpoints)
-	fmt.Println("----------------------------------------------")
-
-	for zone, traffic := range lastResult.TrafficDetail {
-		fmt.Printf("Total to %s \t %.f%% \n", zone, traffic.Incoming*100)
-	}
-	fmt.Println("----------------------------------------------")
-
-	for zone, traffic := range lastResult.TrafficDetail {
-		fmt.Printf("From %s : \n", zone)
-		for z, t := range traffic.Outgoing {
-			fmt.Printf("\t to %s : %.2f%% \n", z, t*100)
-		}
-	}
-	fmt.Println("----------------------------------------------")
-
-	for zone, workload := range lastResult.Workload {
-		fmt.Printf("Workload for %s : \t %.2f%% \n", zone, workload*100)
+// StartSimulation based on the zones(Region) and EndpointSliceGroups
+func (m *Model) StartSimulation() error {
+	simResult, err := m.simulator.Simulate(m.region, m.slices)
+	if err != nil {
+		return err
 	}
+	m.results = append(m.results, simResult)
 	return nil
 }
 
-// GetEndpointSliceGroups returns the EndpointSliceGroups, don't allow users to
-// manually change this field -- use this getter to get the value
-func (m *Model) GetEndpointSliceGroups() map[string]EndpointSliceGroup {
-	return m.slices
-}
-
 // GetNumberOfEndpointSlices returns the number of EndpointSlices
 func (m *Model) GetNumberOfEndpointSlices() uint {
 	totalSlices := uint(0)
 	for _, slice := range m.slices {
-		pods := slice.numberOfPods()
+		pods := slice.numberOfEndpoints()
 		totalSlices += uint(pods) / m.SliceCapacity
 		if uint(pods)%m.SliceCapacity != 0 {
 			totalSlices++
 		}
 	}
 	return totalSlices
 }
+
+// GetLastResult gets the last result of the simulation
+func (m *Model) GetLastResult() SimulationResult {
+	return m.results[len(m.results)-1]
+}
```

Comment:
Is this still being used? I thought we were aggregating results + printing out each individual result?

Reply:
changed the func signature that currently the StartSimulation will return the result. no need to call this separately.